chatgpt-shell-openai.el

```diff
@@ -268,34 +263,35 @@ or
 
 (cl-defun chatgpt-shell-openai--make-payload (&key model context settings)
   "Create the API payload using MODEL CONTEXT and SETTINGS."
-  (chatgpt-shell-openai-make-chatgpt-request-data
+  (apply
+   #'chatgpt-shell-openai-make-chatgpt-request-data
    :system-prompt (map-elt settings :system-prompt)
    :context context
    :version (map-elt model :version)
    :temperature (map-elt settings :temperature)
-   :streaming (map-elt settings :streaming)))
+   :streaming (map-elt settings :streaming)
+   :other-params (map-elt model :other-params)))
 
-(cl-defun chatgpt-shell-openai--handle-chatgpt-command (&key model command context shell settings)
+(cl-defun chatgpt-shell-openai--handle-chatgpt-command (&key model command context shell settings (key #'chatgpt-shell-openai-key) (filter #'chatgpt-shell-openai--filter-output))
   "Handle ChatGPT COMMAND (prompt) using MODEL, CONTEXT, SHELL, and SETTINGS."
-  (unless (chatgpt-shell-openai-key)
+  (unless (funcall key)
     (funcall (map-elt shell :write-output) "Your chatgpt-shell-openai-key is missing")
     (funcall (map-elt shell :finish-output) nil))
   (shell-maker-make-http-request
    :async t
-   :url (concat chatgpt-shell-api-url-base
-                (or (map-elt model :path)
-                    (error "Model :path not found")))
+   :url (chatgpt-shell-openai--make-url :model model)
    :proxy chatgpt-shell-proxy
    :data (chatgpt-shell-openai-make-chatgpt-request-data
           :prompt command
           :system-prompt (map-elt settings :system-prompt)
           :context context
           :version (map-elt model :version)
           :temperature (map-elt settings :temperature)
-          :streaming (map-elt settings :streaming))
+          :streaming (map-elt settings :streaming)
+          :other-params (map-elt model :other-params))
```

Comment:
:other params kinda works, but I backed from previous implementations. We can keep it, but we should consider replacing in the future with a function parameter that's passed the payload, which returns an enhanced payload (with the other parameters added).

Reply:
Yeah, it's a bit ugly though it works. While I was writing this I actually felt like it might be cleaner and easier to extend if the whole thing was refactored to use defclass but that would probably be a lot more work than it would be worth. 

Are you talking about adding a new keyword argument or do you mean using the current :payload with a different value?

Reply:
:finalize (lambda (payload)
  ... add other params here
  return enhanced payload
 )

The benefit is that concrete implementations can choose exactly where the new param is added (not just top level).

Reply:
I'm closing anyway. Future improvement.